lib/validators.ts

```diff
@@ -36,20 +46,37 @@ export const generateFlexibleSchema = (
       if (field.type === 'checkboxes') {
         shape[field.id] = (shape[field.id] as Yup.NumberSchema).min(
           1,
-          field.error || 'Please choose at least one option'
+          getErrorMessage(field)
         );
       } else if (field.type === 'repeater' || field.type === 'repeaterGroup') {
         shape[field.id] = (shape[field.id] as Yup.NumberSchema).min(
           1,
-          field.error || `Please add at least one ${field.itemName || 'item'}`
+          getErrorMessage(field)
         );
       } else {
-        shape[field.id] = shape[field.id].required(
-          field.error || 'This question is required'
-        );
+        shape[field.id] = shape[field.id].required(getErrorMessage(field));
       }
     }
   });
 
   return Yup.object().shape(shape);
 };
+
+/** respect the "required" attribute for conditional fields, only when the condition is met */
+export const validateConditionalFields = (
+  values: FormikValues,
+  fields: Field[]
+): FormikErrors<FormikValues> => {
+  const errors: FormikErrors<FormikValues> = {};
+  fields.map((field) => {
+    if (
+      field.condition &&
+      values[field.condition.id] === field.condition.value &&
+      field.required &&
+      !values[field.id]
```

Comment:
we're just testing for a falsy value (eg. empty string, empty array). are there situations where this won't work?